### PR TITLE
Remove legacy workflow.xml/yaml references and complete workflow.md migration

### PR DESCRIPTION
**Pull request by dickymoore**
## What
Migrated workflows to the Markdown-based `workflow.md` format and removed legacy XML/YAML workflow artifacts and references across the repo. This includes CLI/template/docs/test updates to keep tooling aligned with the single workflow definition format.

## Why
To align with the v6 step-file workflow architecture and eliminate format drift that caused stale references and inconsistent behavior in tooling and workflows. This consolidation reduces maintenance and avoids mismatched file expectations.

## How
- Converted remaining workflows (including QA automate) to `workflow.md` and removed `workflow.yaml`/`workflow.xml` artifacts.
- Updated CLI/IDE tooling and templates to treat `workflow.md` as the canonical workflow definition format.
- Updated workflow reference guards and cleaned references in agents/docs/tests to prevent legacy formats from reappearing.

## Testing
Ran `npm test` (test:schemas, test:install, validate:schemas, lint, lint:md, format:check).

## Summary
Migrated workflows to the Markdown-based format and cleaned up legacy XML/YAML references across the repo, including CLI/template/docs/test updates that align with the new workflow.md structure.

## Detailed Changes
- Removed legacy workflow XML files and associated references.
- Completed migration to `workflow.md` and eliminated `workflow.yaml` references (including the QA automate workflow).
- Updated CLI installer comments/wording to reflect workflow definition files.
- Adjusted templates/docs to align with the workflow.md format.
- Updated tests to remove legacy exclusions for deleted workflow files and tighten reference checks.

## Scope/Overlap Check
- Primary areas: `CHANGELOG.md`, `docs/**`, `src/bmm/agents/**`, `src/bmm/module-help.csv`, `src/bmm/workflows/**`, `src/core/tasks/**`, `src/core/workflows/**`, `src/core/resources/excalidraw/**`, `src/utility/agent-components/**`, `test/test-installation-components.js`, `tools/cli/installers/**`.
- No unrelated cleanup noted beyond the workflow.md migration/removal work.

## Potential Conflicts
- Likely overlap with concurrent CLI/IDE template changes in `tools/cli/installers/lib/ide/**` and workflow manifest generation in `tools/cli/installers/lib/core/**`.
- Recommend rebasing on `upstream/main` before merge to pick up recent CLI/IDE template updates.

## Migration Note
- Only `workflow.md` is supported; legacy `workflow.xml`/`workflow.yaml` files should be converted to Markdown-based workflows.

## Compatibility Notes
- Workflow status YAML references (`bmm-workflow-status.yaml` / `workflow-status`) are preserved; status tracking behavior not changed.

## Commit Breakdown
- Make workflow references platform-agnostic.
- Convert sprint-planning + sprint-status workflows to MD.
- Convert Phase 1 workflows to MD; add advanced-elicitation MD workflow + guard.
- Convert Phase 2/3 workflows to MD.
- Add validate-workflow MD task and guard.
- Migrate workflow runner references to workflow.md.
- Remove workflow.xml runner and XML workflow file(s).
- Drop YAML workflow support in CLI tooling and remove YAML templates.
- Remove workflow.yaml mention from changelog.
- Remove unused workflow command template.

## Notes / Breaking Changes
- Legacy `workflow.xml` and `workflow.yaml` artifacts are no longer supported; workflows should use `workflow.md`.

## Note for maintainers:
  - This is a substantial refactor, so it may be presumptuous. I hope it’s helpful; if it’s not the direction you want, I’m happy to close it or rework it to better match the project’s needs.

## Tracking
Refs #1585

### CODE REVIEW

**Comment by dickymoore** [comment]
@bmadcode thanks for the comment, and for the invitation to help. I'd love to help out if I can. I'll continue working through this, and will make the PR as draft until it's ready for another review.

**Comment by dickymoore** [comment]
@coderabbitai review

**Comment by dickymoore** [comment]
Process update (clean):

- Added tracking issue linkage in PR body: Refs #1585 (large-change tracking).
- Re-ran validation locally after latest fixes: npm test, npm run validate:refs, plus fresh install checks (tools: none, claude-code, windsurf) all passed.
- Addressed previously reported technical blockers around:
  - workflow runner install path
  - workflow-*.md manifest coverage
  - template selection behavior
  - path/regex handling in installers
  - remaining step-flow conversions

If preferred, this can be split for easier review due size, I can break follow-on cleanup into smaller PRs and keep this PR strictly migration-critical.

File: test/test-installation-components.js [hunk 1]
```diff
@@ -186,6 +186,135 @@ async function runTests() {
...
+    }
+
+    assert(
+      offenders.length === 0,
+      'No validate-workflow.xml references outside XML source',
+      offenders.length > 0 ? offenders.join(', ') : '',
+    );
+  } catch (error) {
+    assert(false, 'Validate workflow reference guard runs', error.message);
+  }
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply test/test-installation-components.js:380]: This exclusion path is now redundant for the current scan roots, so I  removed the unused exclusion wiring and kept the existing guard behavior unchanged.

File: src/bmm/workflows/2-plan-workflows/create-ux-design/steps/step-04-emotional-response.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ This step will generate content and present choices:
 
 ## PROTOCOL INTEGRATION:
 
-- When 'A' selected: Read fully and follow: {project-root}/_bmad/core/workflows/advanced-elicitation/workflow.xml
+- When 'A' selected: Read fully and follow: {project-root}/_bmad/core/workflows/advanced-elicitation/workflow.md
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/2-plan-workflows/create-ux-design/steps/step-04-emotional-response.md:33]: The excluded-file path was redundant for the current scan roots, so I  removed the unused exclusion wiring while preserving the same guard behaviour.   Verified by test suite.

File: src/bmm/workflows/4-implementation/code-review/steps/step-03-execute-review.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        - Edge cases and null handling
+        - Architecture violations
+        - Documentation gaps
+        - Integration issues
+        - Dependency problems
+        - Git commit message quality (if applicable)
+      </action>
+      <action>Find at least 3 more specific, actionable issues</action>
+    </check>
+  </step>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/4-implementation/code-review/steps/step-03-execute-review.md:155]: Step 3 now defines a required findings schema (id, severity, type, summary, detail, file_line, proof, suggested_fix, reviewer, timestamp), maps how each review action populates those fields, persists findings to {story_dir}/review-findings.json, and includes an example record. Step 4 now loads that artifact and validates the schema before use

File: src/bmm/workflows/4-implementation/code-review/steps/step-03-execute-review.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    <!-- Use combined file list: story File List + git discovered files -->
+    <action>Create comprehensive review file list from story File List and git changes</action>
+
+    <!-- AC Validation -->
+    <action>For EACH Acceptance Criterion:
+      1. Read the AC requirement
+      2. Search implementation files for evidence
+      3. Determine: IMPLEMENTED, PARTIAL, or MISSING
+      4. If MISSING/PARTIAL → HIGH SEVERITY finding
+    </action>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/4-implementation/code-review/steps/step-03-execute-review.md:100]: the “Determine: IMPLEMENTED, PARTIAL, or MISSING” section now includes clause-level criteria, evidence-strength rules, indirect-evidence handling, explicit severity mapping (HIGH/MEDIUM/LOW), and classification examples.

File: src/bmm/workflows/4-implementation/correct-course/workflow.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
+name: correct-course
+description: "Navigate significant changes during sprint execution by analyzing impact, proposing solutions, and routing for implementation"
+main_config: '{project-root}/_bmad/bmm/config.yaml'
+web_bundle: false
+---
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/4-implementation/correct-course/workflow.md:35]: Installed runtime path is preserved, and a source-path fallback resolver was added. The workflow now resolves to installed_path first, falls back to src path when needed, and HALTs with a clear error if neither path available.

**Comment by dickymoore** [comment]
@coderabbitai review

File: src/core/tasks/validate-workflow.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+## Initialization
+- Load config from `{project-root}/_bmad/core/config.yaml`.
+- Validate config load before continuing:
+  - Verify file exists and is readable.
+  - Parse YAML and fail fast with explicit error if parsing fails.
+  - Require `user_name`; if missing, abort initialization with descriptive error.
+  - Apply explicit defaults when optional keys are absent:
+    - `communication_language = "en"`
+    - `document_output_language = "en"`
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/core/tasks/validate-workflow.md]: updated fallback defaults to match src/core/module.yaml (communication_languageanddocument_output_language now default to "English"), keeping the same variable symbols used by templates.

File: tools/cli/installers/lib/ide/templates/combined/kiro-workflow-yaml.md [hunk 7]
```diff
@@ -7,9 +7,9 @@ inclusion: manual
 IT IS CRITICAL THAT YOU FOLLOW THESE STEPS - while staying in character as the current agent persona you may have loaded:
 
 <steps CRITICAL="TRUE">
-1. Always LOAD the FULL #[[file:{{bmadFolderName}}/core/tasks/workflow.xml]]
+1. Always LOAD the FULL #[[file:{{bmadFolderName}}/core/tasks/workflow.md]]
 2. READ its entire contents - this is the CORE OS for EXECUTING the specific workflow-config #[[file:{{bmadFolderName}}/{{path}}]]
-3. Pass the yaml path {{bmadFolderName}}/{{path}} as 'workflow-config' parameter to the workflow.xml instructions
-4. Follow workflow.xml instructions EXACTLY as written to process and follow the specific workflow config and its instructions
+3. Pass the workflow path {{bmadFolderName}}/{{path}} as 'workflow-config' parameter to the workflow.md task instructions
+4. Follow workflow.md task instructions EXACTLY as written to process and follow the specific workflow config and its instructions
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply tools/cli/installers/lib/ide/templates/combined/kiro-workflow-yaml.md]: Clarified the exact parameter mechanism for workflow invocation by specifying YAML key workflow-configand adding an explicit example block with value{{bmadFolderName}}/{{path}}.

**Comment by dickymoore** [comment]
I hope this is helpful. I tried to progress this work as much as I could. There are lots of changes over many commits in response to coderabbitai and local adversarial reviews. 

## What I fixed
- Restored empty-target short-circuit handling in `tools/cli/installers/lib/ide/_config-driven.js` (`artifact_types: []` no longer creates empty output directories).
- Restored split-template extension override behavior in `tools/cli/installers/lib/ide/_config-driven.js` (no forced `.md`; extension normalization added).
- Restored defensive workflow path normalization in `tools/cli/installers/lib/ide/shared/workflow-command-generator.js` (separator normalization + `/src/...`fallback mapping before `_bmad/` extraction).
- Removed stale XML doc reference and tightened user-facing wording in migrated workflow/docs content.

## Tests/validation
- `npm test` passes.
- Added regression guards in `test/test-installation-components.js`:
  - empty artifact target short-circuit
  - split-template extension override
  - workflow path normalization

## PR size
- PR is still large (`129 files`, `+3327/-2176`). If necessary, I could try breaking this up into multiple PRs, and would love to help in other ways if there's extra guidance for you to offer @bmadcode.

Thanks, Dicky

File: test/test-installation-components.js [hunk 8]
```diff
@@ -186,6 +186,135 @@ async function runTests() {
...
+    };
+
+    for (const root of searchRoots) {
+      await walk(root);
+    }
+
+    assert(offenders.length === 0, 'No workflow.xml references outside XML source', offenders.length > 0 ? offenders.join(', ') : '');
+  } catch (error) {
+    assert(false, 'Workflow reference guard runs', error.message);
+  }
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply test/test-installation-components.js:410]: The guard now uses a filename-aware regex (`forbiddenRefPattern`) instead of raw substring matching, so `validate-workflow.xml` no longer false- positives as `workflow.xml`. I also kept the offenders list behavior unchanged and expanded scan coverage to installer JS files.

File: tools/cli/installers/lib/modules/manager.js [hunk 9]
```diff
@@ -757,90 +756,33 @@ class ModuleManager {
...
-          break;
-        }
-      }
-
-      // If we didn't find an end, it goes to end of file
-      if (endIdx === startIdx) {
-        endIdx = lines.length - 1;
-      }
+    await fs.writeFile(targetFile, mdContent, 'utf8');
+  }
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply tools/cli/installers/lib/modules/manager.js:791]: Removed the no-op `_bmad -> _bmad` replacement in `copyWorkflowFileStripped`; only the real placeholder substitution (`_bmad -> this.bmadFolderName`) remains before frontmatter stripping/write.

File: src/bmm/workflows/4-implementation/dev-story/steps/step-08-mark-task-complete.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+      <!-- Mark task in Review Follow-ups section -->
+      <action>Mark task checkbox [x] in "Tasks/Subtasks → Review Follow-ups (AI)" section</action>
+
+      <!-- CRITICAL: Also mark corresponding action item in review section -->
+      <action>Find matching action item in "Senior Developer Review (AI) → Action Items" section by matching description</action>
+      <action>Mark that action item checkbox [x] as resolved</action>
+
+      <action>Add to Dev Agent Record → Completion Notes: "✅ Resolved review finding [{{severity}}]: {{description}}"</action>
+    </check>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/4-implementation/dev-story/steps/step-08-mark-task-complete.md:84]: `resolved_review_items` / `unresolved_review_items` are now initialized at the start of step 8. I also added explicit fuzzy matching logic (normalize, exact/substring pass, token-set Jaccard scoring, threshold, tie handling, and unmatched handling) and deterministic recording behavior.

File: src/bmm/workflows/4-implementation/dev-story/steps/step-08-mark-task-complete.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+    <check if="ANY validation fails">
+      <action>DO NOT mark task complete - fix issues first</action>
+      <action>HALT if unable to fix validation failures</action>
+    </check>
+
+    <check if="review_continuation == true and {{resolved_review_items}} is not empty">
+      <action>Count total resolved review items in this session</action>
+      <action>Add Change Log entry: "Addressed code review findings - {{resolved_count}} items resolved (Date: {{date}})"</action>
+    </check>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/4-implementation/dev-story/steps/step-08-mark-task-complete.md:106]: Step 8 now initializes the missing variables before use: `review_continuation`, `resolved_count`, and `date` (system-generated timestamp formatting). The changelog/update checks now run against initialized values.

File: src/bmm/workflows/document-project/instructions.md [hunk 12]
```diff
@@ -10,10 +10,26 @@
...
+<action>Attempt to load workflow status directly from `{output_folder}/bmm-workflow-status.yaml`:
+  - If file exists, is readable, and parses correctly:
+    - Set status_exists = true
+    - Set status_file_found = true
+    - Set standalone_mode = false
+    - Set status_file_path = `{output_folder}/bmm-workflow-status.yaml`
+    - Extract field_type, warning, suggestion, next_workflow, next_agent if present
+  - If file is missing, unreadable, or malformed:
+    - Keep defaults and continue in standalone mode
+</action>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/document-project/instructions.md:43]: Status-load fallback now captures and outputs explicit failure context (`status_load_error_reason`) and emits a warning message including the attempted `bmm-workflow-status.yaml` path.

File: src/bmm/workflows/document-project/instructions.md [hunk 13]
```diff
@@ -58,12 +74,23 @@
 <critical>SMART LOADING STRATEGY: Check state file FIRST before loading any CSV files</critical>
 
 <action>Check for existing state file at: {output_folder}/project-scan-report.json</action>
+<action>Set resume_mode = false</action>
 
 <check if="project-scan-report.json exists">
   <action>Read state file and extract: timestamps, mode, scan_level, current_step, completed_steps, project_classification</action>
   <action>Extract cached project_type_id(s) from state file if present</action>
   <action>Calculate age of state file (current time - last_updated)</action>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/document-project/instructions.md:108]: Added explicit `last_updated` validation and stale fallback pathing: invalid/missing timestamps set the stale age sentinel, and age calculation now uses a validated timestamp variable.

File: src/bmm/workflows/document-project/instructions.md [hunk 14]
```diff
@@ -86,6 +113,7 @@ Your choice [1/2/3]:
   <check if="user selects 1">
     <action>Set resume_mode = true</action>
     <action>Set workflow_mode = {{mode}}</action>
+    <action>Set subworkflow_success = false</action>
     <action>Load findings summaries from state file</action>
     <action>Load cached project_type_id(s) from state file</action>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/document-project/instructions.md:161]: Resume mode now validates persisted `mode` before assigning `workflow_mode`; invalid mode falls back to a safe fresh-scan path instead of trusting state blindly.

File: src/bmm/workflows/document-project/instructions.md [hunk 15]
```diff
@@ -157,36 +213,60 @@ Your choice [1/2/3]:
...
+    <output>Invalid selection. Expected 1, 2, or 3. Exiting workflow without changes.</output>
+    <action>Exit workflow</action>
+  </check>
 </check>
 
 <check if="index.md does not exist">
   <action>Set workflow_mode = "initial_scan"</action>
   <action>Display: "No existing documentation found. Starting initial project scan..."</action>
   <action>Read fully and follow: {installed_path}/workflows/full-scan-instructions.md</action>
+  <action>Set subworkflow_success = true only if delegated workflow completed without HALT/error</action>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/document-project/instructions.md:303]: Initial-scan branch now sets `scan_level` explicitly before delegation, so downstream output/state is never undefined in that path.

File: src/bmm/workflows/document-project/instructions.md [hunk 16]
```diff
@@ -196,25 +276,21 @@ Your choice [1/2/3]:
...
-
-- Progress tracking updated
+<check if="status_file_found == true AND status_update_success == true">
+  <output>**Status Updated:** Progress tracking updated.
 
 **Next Steps:**
-
 - **Next required:** {{next_workflow}} ({{next_agent}} agent)
+- Run `bmad-help` if you need recommended next workflows.</output>
+</check>
```
[the remark this answers is not in the document]
Comment (dickymoore) [reply src/bmm/workflows/document-project/instructions.md:356]: “Next Steps” output is now guarded: it renders `next_workflow`/ `next_agent` only when both are present and otherwise falls back to `not specified`, preventing blank placeholders.

**Comment by dickymoore** [comment]
Regarding the issue of this being a very large PR, here is how it could be split into multiple PRs which gives reviewable chunks, each green and mergeable, without breaking runtime behaviour. If maintainers can give me guidance on whether that would be helpful or not, I'll act accordingly. 

Here's the plan:

1. Create fresh branches from upstream/main.
2. Pull in changes by file scope (git restore --source phase1-md-workflows-clean --<paths>), not by old commit history.
3. Commit each scope as its own PR.
4. Run full checks per PR (npm test, npm run validate:refs).

## PRs: 

###  1. PR A: CLI/runtime migration foundation

  - tools/cli/installers/**
  - src/utility/agent-components/**
  - src/core/tasks/workflow.md, src/core/tasks/validate-workflow.md
  - test/test-installation-components.js, test/test-file-refs-csv.js

###  2. PR B: Core workflow/task format clean-up

  - src/core/workflows/**
  - remove legacy XML runner/file references
  - keep only intentional YAML config files (e.g. document-project subworkflow configs)

###  3. PR C: BMM Phase 1–3 step-file migration

  - src/bmm/workflows/1-analysis/**
  - src/bmm/workflows/2-plan-workflows/**
  - src/bmm/workflows/3-solutioning/**
  - src/bmm/module-help.csv updates tied to those workflows

###  4. PR D: BMM Phase 4 implementation workflows

  - src/bmm/workflows/4-implementation/**
  - includes code-review/dev-story/create-story/correct-course contract fixes

###  5. PR E: Docs and onboarding alignment

  - docs/**, README.md, command/tutorial docs
  - only after behaviour/paths are already settled

**Comment by bmadcode** [comment]
Hey @dickymoore sorry for the late reply, have been a bit under the weather the past few days, but doing better now will have time to look at the changes and suggestion. Will respond more in the morning after I can read through the proposal and better answer the questions. Appreciate this!

**Comment by dickymoore** [comment]
> Hey @dickymoore sorry for the late reply, have been a bit under the weather the past few days, but doing better now will have time to look at the changes and suggestion. Will respond more in the morning after I can read through the proposal and better answer the questions. Appreciate this!

Sorry to hear you've been under the weather. I've been knocked over by a virus for the last few days too. Winter is hitting. Apologies this PR is so huge. No worries I need to try something else; if it's implemented the wrong way or isn't the direction you wanted to go in - I'd be happy to try a different tack. Will follow your guidance.

**Comment by bmadcode** [comment]
> > Hey @dickymoore sorry for the late reply, have been a bit under the weather the past few days, but doing better now will have time to look at the changes and suggestion. Will respond more in the morning after I can read through the proposal and better answer the questions. Appreciate this!
> 
> Sorry to hear you've been under the weather. I've been knocked over by a virus for the last few days too. Winter is hitting. Apologies this PR is so huge. No worries I need to try something else; if it's implemented the wrong way or isn't the direction you wanted to go in - I'd be happy to try a different tack. Will follow your guidance.

I think I am finally almost on the other side of it - good luck for you also getting through it I hope MUCH quicker :)

I think you are on discord - let me ping you there and we can discuss this one a bit easier.

**Comment by bmadcode** [comment]
closing this as discussed, will align on plan in discord @dickymoore
